packages/node_modules/@node-red/runtime/lib/nodes/flows/Subflow.js

```diff
@@ -192,6 +193,92 @@ class Subflow extends Flow {
         super.start(diff);
     }
 
+    /**
+     * Get environment variable of subflow
+     * @param {String}   name   name of env var
+     * @return {Object}  val    value of env var
+     */
+    getSetting(name) {
+        function lookup(env, name) {
+            for(var i = 0; i < env.length; i++) {
+                var item = env[i];
+                if (item.name === name) {
+                    return item;
+                }
+            }
+            return null;
+        }
+        function lookupFlow0(flow, name) {
+            if (flow.subflowInstance && flow.subflowInstance.env) {
+                var val = lookup(flow.subflowInstance.env, name);
+                if (val) {
+                    return val;
+                }
+            }
+            if (flow.subflowDef && flow.subflowDef.env) {
+                var val = lookup(flow.subflowDef.env, name);
+                if (val) {
+                    return val;
+                }
+            }
+            return null;
+        }
+        function lookupFlow(flow, name) {
+            var prefix = name.substring(0, name.length -5); // len("_{type,info}")==5
```

Comment:
I'm not clear what this is doing or where it gets used.

My expectation is the type information is held by the subflow and used *by the subflow* to return the env-var already converted to the appropriate type.

Reply:
Current specification allow type and meta data for a variable *V* can be accessed as *V_type* and *V_info*.   *lookupFlow* looks up value of *V_type*, *V_info*, or *V*.
Is this make your concern clear?  Or, am I misunderstanding your concern?

Reply:
When does a user of this need to lookup the type and info meta data? My expectation is `getSetting` returns the value of the property *already converted to the appropriate type*. Otherwise, every single node that uses this api has to implement all the code to get the value, then get its type, then parse the value to the real value according to its type.

Reply:
I think there are two issues here:
1. representation of values,
2. needs for meta-data (*V_type*, *V_info*).

For #1, I'll commit changes that convert values according to specified types.
So, parsing values become unnecessary.
For #2, I think we need meta-data for TypedInput-style parameter input, error handling, ....

Reply:
I don't understand your claim on 2. Yes we need the type meta data against the value - but the only thing that needs it is the Subflow who will use that information to return the value converted to the right type to anyone who asks for it. The _info meta data is out of scope for this PR and I don't want to drag in design discussions about UI generation into what should be a simple runtime feature.

Reply:
My intention to include *_type* and *info* meta-data is consistency with interface with UI definition.  But, there is no problem deleting this feature from this PR and recovering it with UI definition.

Reply:
Updated code according to your comments.